src/main/java/org/opensearch/ad/rest/RestValidateDetectorAction.java

```diff
@@ -1,44 +1,130 @@
 import org.opensearch.ad.constant.CommonErrorMessages;
+import org.opensearch.ad.model.AnomalyDetector;
+import org.opensearch.ad.model.DetectorValidationIssue;
+import org.opensearch.ad.model.ValidationAspect;
+import org.opensearch.ad.settings.AnomalyDetectorSettings;
 import org.opensearch.ad.settings.EnabledSetting;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorAction;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorRequest;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentType;
 import org.opensearch.extensions.rest.ExtensionRestRequest;
 import org.opensearch.extensions.rest.ExtensionRestResponse;
 import org.opensearch.rest.RestHandler.Route;
 import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.rest.action.RestToXContentListener;
 import org.opensearch.sdk.ExtensionRestHandler;
+import org.opensearch.sdk.ExtensionsRunner;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ad.util.RestHandlerUtils.TYPE;
+import static org.opensearch.ad.util.RestHandlerUtils.VALIDATE;
+import static org.opensearch.rest.RestStatus.*;
 
 public class RestValidateDetectorAction implements ExtensionRestHandler {
     private final Logger logger = LogManager.getLogger(RestValidateDetectorAction.class);
+    private final OpenSearchClient sdkClient;
+    private final NamedXContentRegistry xContentRegistry;
 
+    public static final Set<String> ALL_VALIDATION_ASPECTS_STRS = Arrays
+            .asList(ValidationAspect.values())
+            .stream()
+            .map(aspect -> aspect.getName())
+            .collect(Collectors.toSet());
+    public RestValidateDetectorAction(ExtensionsRunner runner, AnomalyDetectorExtension extension){
+        this.xContentRegistry = runner.getNamedXContentRegistry().getRegistry();
+        this.sdkClient = extension.getClient();
+    }
     @Override
     public List<Route> routes() {
         return List.of(new Route(POST, "/detectors/_validate"), new Route(POST, "/detectors/_validate/{type}"));
     }
 
     @Override
-    public ExtensionRestResponse handleRequest(ExtensionRestRequest request) {
+    public ExtensionRestResponse handleRequest(ExtensionRestRequest request){
         if (!EnabledSetting.isADPluginEnabled()) {
             throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
         }
-        Method method = request.method();
-
-        if (!Method.POST.equals(method)) {
-            return new ExtensionRestResponse(
-                request,
-                NOT_FOUND,
-                "Extension REST action improperly configured to handle " + request.toString()
-            );
```

Comment:
Why are we removing this check here?

Reply:
yeah  I think we can keep it as initially I thought to be a template just to fill the method. But, as this is a check to validate method as POST we can keep that.

Reply:
We could remove the method check if we use the new `routeHandlers()` syntax.

Reply:
ACK